### PR TITLE
Remove app suffix requirement from template-app

### PR DESCRIPTION
**Pull request by cokiengchiara**
Towards https://github.com/giantswarm/giantswarm/issues/9815

1. I also added template for adding basic info about the app from the app issue https://github.com/giantswarm/giantswarm/issues/new?assignees=&labels=area%2Fmanaged-apps%2C+team%2Fhalo&template=new-app-in-catalog-preview.md&title=
2. I also added the sample readme outline from https://intranet.giantswarm.io/docs/dev-and-releng/app-developer-processes/apps_in_happa/

The idea is they are easy to fill out while the App Engineer is adding an app. Do you think this is reasonable to include in the template?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,10 +1,62 @@
-[![CircleCI](https://circleci.com/gh/giantswarm/{APP-NAME}-app.svg?style=shield)](https://circleci.com/gh/giantswarm/{APP-NAME}-app)
+[![CircleCI](https://circleci.com/gh/giantswarm/{APP-NAME}.svg?style=shield)](https://circleci.com/gh/giantswarm/{APP-NAME})
```
Comment (puja108) [inline README.md:1]: this change I would remove, cause in the repo name it's ok

File: helm/APP-NAME-app/Chart.yaml [hunk 2]
```diff
@@ -2,9 +2,9 @@ apiVersion: v1
 appVersion: 0.0.1
 description: A Helm chart for {APP-NAME}
 engine: gotpl
-home: https://github.com/giantswarm/{APP-NAME}-app
+home: https://github.com/giantswarm/{APP-NAME}
```
Comment (puja108) [inline helm/APP-NAME-app/Chart.yaml:5]: I'd keep the name

File: README.md (hunk 1 above)
Comment (cokiengchiara) [inline README.md:1]:
```suggestion
[![CircleCI](https://circleci.com/gh/giantswarm/{APP-NAME}.svg?style=shield)](https://circleci.com/gh/giantswarm/{APP-NAME}-app)
```

File: helm/APP-NAME-app/Chart.yaml (hunk 2 above)
Comment (cokiengchiara) [inline helm/APP-NAME-app/Chart.yaml:5]:
```suggestion
home: https://github.com/giantswarm/{APP-NAME}-app
```

Comment (cokiengchiara) [inline helm/APP-NAME-app/Chart.yaml]:
```suggestion
home: https://github.com/giantswarm/{APP-NAME}-app
```

File: README.md [hunk 3]
```diff
@@ -1,10 +1,62 @@
...
+```
+# values.yaml
+
+```
+
+### Sample App CR and ConfigMap for the Control Plane
+If you have access to the Kubernetes API on the Control Plane, you could create
+the App CR and ConfigMap directly.
+
+Here is an example that would install the nginx-ingress-controller-app to
```
Comment (cokiengchiara) [inline README.md:33]:
```suggestion
Here is an example that would install the app to
```

File: README.md (hunk 1 above)
Comment (cokiengchiara) [reply README.md]: removed the change.

File: helm/APP-NAME-app/Chart.yaml (hunk 2 above)
Comment (cokiengchiara) [reply helm/APP-NAME-app/Chart.yaml]: OK kept the name.

File: README.md [hunk 4]
```diff
@@ -1,10 +1,62 @@
...
+### Sample App CR and ConfigMap for the Control Plane
+If you have access to the Kubernetes API on the Control Plane, you could create
+the App CR and ConfigMap directly.
+
+Here is an example that would install the app to
+tenant cluster `abc12`:
+
+```
+# appCR.yaml
+
```
Comment (webwurst) [inline README.md:38]: Still something missing here?

**Review by puja108 — APPROVED** [review]

File: README.md (hunk 4 above)
Comment (cokiengchiara) [reply README.md:38]: I expect the developer to fill this with an example 🙈 it's not clear?

Comment (webwurst) [reply README.md:38]: Ah ok, clear now ;)

**Comment by cokiengchiara** [comment]
@puja108 testes failed on CircleCI... is that fine because this is only a template?

**Comment by cokiengchiara** [comment]
Confirmed that test will not work for template. merging.

**Pull request merged**